biosys/apps/main/api/uploaders.py

```diff
@@ -362,11 +378,44 @@ def schema(self):
     def fields(self):
         return self.schema.get('fields') if self.schema else []
 
+    def get_fields_by_name(self, name):
+        return [f for f in self.fields if f.get('name') == name]
+
     def _biosys_infer(self):
         """
         Rules:
-        - fields of type 'any' should be converted to type 'string'
+        - Fields of type 'any' should be converted to type 'string'
+        - Fields of type 'date' or 'datetime' should have format = 'any' instead of default
+          (the 'any' makes the date parser more flexible)
+        - If it contains a latitude/longitude schema set the lat/long columns type='number' with constraint required
+          and they should be tagged with the correct biosys tag.
         """
-        for field in [f for f in self.fields if f.get('type') == 'any']:
-            field['type'] = 'string'
+        for field in self.fields:
+            type_ = field.get('type')
+            format_ = field.get('format')
+
+            if type_ == 'any':
+                field['type'] = 'string'
+
+            if type_ in ['date', 'datetime'] and format_ == 'default':
+                field['format'] = 'any'
+
+        # geometry inference
+        geo_parser = GeometryParser(self.schema)
+        if geo_parser.is_lat_long:
+            lat_fields = self.get_fields_by_name(geo_parser.latitude_field.name)
+            lon_fields = self.get_fields_by_name(geo_parser.longitude_field.name)
+            if len(lat_fields) == 1 and len(lon_fields) == 1:
+                lat_field = lat_fields[0]
+                lon_field = lon_fields[0]
+                self.set_type('number', lat_field)
+                self.set_type('number', lon_field)
+                self.set_required(lat_field)
+                self.set_required(lon_field)
+                self.set_biosys_type(lat_field, BiosysSchema.LATITUDE_TYPE_NAME)
+                self.set_biosys_type(lon_field, BiosysSchema.LONGITUDE_TYPE_NAME)
+            else:
+                # more that one lat or long fields? Should not happen.
+                pass
```

Comment:
Are you planning on putting some error handling here? if not maybe remove else/pass block

Reply:
There's a basic error handler. I've added an error.